src/functions/sendCandidateResults/application/service/send-notification.ts

```diff
@@ -0,0 +1,109 @@
+import { DocumentsServiceError } from '../../domain/errors/documents-service-error';
+import { PersonalisationDetails } from '../../domain/personalisation.model';
+import { INotifyClient } from '../../domain/notify-client.interface';
+import { get } from 'lodash';
+import { AxiosError } from 'axios';
+import * as Handlebars from 'handlebars';
+import { subjectMapper, templateMapper } from './template-selector';
+import { Language } from '../../domain/conducted-language';
+import { TestOutcome } from '../../domain/test-outcome';
+import { CommunicationMethod } from '@dvsa/mes-test-schema/categories/common';
+import { Correspondence } from '../../domain/template-id.model';
+
+export async function sendNotification(
+  emailAddress: string,
+  communicationMethod: CommunicationMethod,
+  templateId: string,
+  notificationPersonalisation: PersonalisationDetails,
+  reference: string,
+  emailReplyToId: string,
+  client: INotifyClient,
+  conductedLanguage: Language,
+  testOutcome: TestOutcome,
+  padi?: boolean,
+): Promise<any> {
+
+  try {
+    let address;
```

Comment:
This can be move to after the first try/catch,  we don't yet know if this will be set - can we also please add appropriate types.